File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-0.py
class Solution:
    def findMin(self, nums: List[int]) -> int:
        # okay die Idee ist hier dadurch das Ding vom kleinsten Element aus verschoben ist muss eine 
        # eine Seite sortiert sein udn die ndere nicht das gesuche elemten ist also in der andere n unsortierten Siete


        # fangen wir wie bei Binary seach an 
        l = 0
        r =len(nums) -1

        while l < r:
            m = l +(r-l)//2
            if nums[m] < nums[r]:
                r = m
            else:
                l = m +1
        return nums[l]

File: Data Structures & Algorithms/three-integer-sum/submission-0.cs
public class Solution {
    public List<List<int>> ThreeSum(int[] nums) 
    {
        Array.Sort(nums);
        List<List<int>> resList = new List<List<int>>(); 
        int k =nums.Length -1;
        
        for(int i = 0; i < nums.Length ; i++)
        {
        int j = i+1;
        
        while( j < k )
        {

        if(nums[i] + nums[j]+ nums[k] == 0) 
        {
        resList.Add(new List<int> {nums[i],nums[j],nums[k]});
        j++; 
        k--; 
        }
        else if ( nums[i] + nums[j]+ nums[k] < 0 ) j++;
        else k--;
        

        }

        }
    return resList; 
    }
     
        
    }

